third-party/gmp/Makefile

```diff
@@ -11,11 +11,24 @@ include $(CHPL_MAKE_HOME)/make/Makefile.base
 GMP_UNPACKED_DIR = gmp-$(GMP_VERSION)
 GMP_TARBALL = $(GMP_UNPACKED_DIR)a.tar.bz2
 
+#
+# Cray X* builds are cross-compilations, as are Intel Phi KNC builds.
+#
+GMP_CROSS_COMPILED=no
+ifneq (, $(filter cray-x%,$(CHPL_MAKE_TARGET_PLATFORM)))
+CHPL_GMP_CFG_OPTIONS += --build=x86_64-unknown-linux-gnu \
+                        --host=x86_64-cnl-linux-gnu
+GMP_CROSS_COMPILED=yes
+else ifeq ($(CHPL_MAKE_TARGET_ARCH),knc)
+CHPL_GMP_CFG_OPTIONS += --host=x86_64-unknown-linux-gnu
```

Comment:
Yet another update! :-)
I've done some more studying, and based on one of the Gnu autoconf web pages, for least fragility we should set both --build and --host. I'll figure out some way to get the right thing to happen with non-Cray KNL.